.github/workflows/hygiene.yml

```
@@ -0,0 +1,15 @@
+name: Hygiene
+
+on: [push, pull_request]
```

```suggestion
on:
  push:
    branches: [ master ]  
  pull_request:
    branches: [ master ]
```

I would prefer explicitly set branches ...

---

the branch is called main in this repo and not master anymore

---

Fixed it ...

---

but why do you want it to be that explicit? Those are linting and unit tests so I think they should be executed on any branch not just the main one.

---

I guess this really depends on the workflow. If you use the forked git workflow that it doesn't make a real difference. If you are using the feature git workflow where feature branches are in the upstream repo it makes sense to not run the ci on these feature branches.

---

I don't follow why doesn't it make sense to run unit tests and linting checks on a feature branch?

---

Even when you work trunk based where the branches are really short living I would like to have that for the case of collaboration within a branch before creating a PR. From that kind of thinking when working with a feature git flow where the branches are relatively long living I think it's even more crucial to run them regularly?

---

Or to elaborate differently as you wrote on forking I don't care because I can just use main within in my own space

When doing feature branch and if CI checking is limited to main branch you're leaving me with two choices: 

a.) create a draft pr immediately and checking the results of the CI within that pr regularly although my main focus is on a feature branch
b.) fly blindly until the PR is ready and the feature is done with a high chance of having to fix a lot of issues at once since it was just forgotten while developing that feature

So I actually think limiting unit tests and linting to main is a counter intuitive choice on each workflow.

---

I am fine with your pro's for using CI for all branches. You do not want to run tests on your machine but just use the CI ...